app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,74 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: [:public_recipe]
+
+  def index
+    @recipes = current_user.recipes.order(created_at: :desc)
+  end
+
+  def show
+    @recipe = Recipe.find_by(id: params[:id])
+    @recipe_foods = RecipeFood.where(recipe_id: params[:id])
+
+    if @recipe_foods.any?
+      @ingredients = []
+      @recipe_foods.each do |recipe_food|
+        @ingredients << {
+          id: recipe_food.id,
+          name: Food.find_by(id: recipe_food.food_id).name,
+          quantity: recipe_food.quantity,
+          measurement_unit: Food.find_by(id: recipe_food.food_id).measurement_unit,
+          price: recipe_food.quantity * Food.find_by(id: recipe_food.food_id).price
+        }
+      end
+    end
+
+    return unless @recipe.nil?
+
+    flash[:error] = 'Recipe not found'
+    redirect_to recipes_path
+  end
+
+  def public_recipe
+    @recipes = Recipe.includes(:user).where(public: true).order(created_at: :desc)
+  end
```

Comment:
- [ ] One of the requirements is that the public recipe should be displayed ordered by the newest in terms of the newest to be made public and not when the recipe was created. Kindly change the logic for this. 👍 😎

Reply:
@henask12